smk/QC_0.smk

```diff
@@ -0,0 +1,515 @@
+#!/usr/bin/env python
+
+'''
+Pre-processing of metaG and metaT data step - quality reads filtering
+
+Authors: Carmen Saenz, Mani Arumugam, Judit Szarvas
+'''
+
+#
+# configuration yaml file
+# import sys
+import re
+from os import path, scandir
+
+
+localrules: initial_fastqc, qc0_fake_move_for_multiqc, qc0_create_multiqc, \
+            qc1_check_read_length_merge, qc1_cumulative_read_len_plot, \
+            qc2_config_yml_file
+
+# Get common config variables
+include: 'include/cmdline_validator.smk'
+include: 'include/config_parser.smk'
+
+if config['raw_reads_dir'] is None:
+    print('ERROR in ', config_path, ': raw_reads_dir variable in configuration yaml file is empty. Please, complete ', config_path)
+else:
+    raw_dir = config['raw_reads_dir']
+
+if config['FASTP_threads'] is None:
+    print('ERROR in ', config_path, ': FASTP_threads variable is empty. Please, complete ', config_path)
+elif type(config['FASTP_threads']) != int:
+    print('ERROR in ', config_path, ': FASTP_threads variable is not an integer. Please, complete ', config_path)
+
+if config['FASTP_memory'] is None:
+    print('ERROR in ', config_path, ': FASTP_memory variable is empty. Please, complete ', config_path)
+elif type(config['FASTP_memory']) != int:
+    print('ERROR in ', config_path, ': FASTP_memory variable is not an integer. Please, complete ', config_path)
+
+if config['FASTP_adapters'] in ('Skip', 'Quality', 'Overlap', 'Detect'):
+    pass
+elif config['FASTP_adapters'] is None:
+    print('ERROR in ', config_path, ': FASTP_adapters variable is empty. Please, complete ', config_path)
+elif path.exists(config['FASTP_adapters']) is False:
+    print('ERROR in ', config_path, ': FASTP_adapters variable path does not exit. Please, complete ', config_path)
+
+# Make list of illumina samples, if ILLUMINA in config
+ilmn_samples = list()
+ilmn_samples_organisation = "folder"
+if 'ILLUMINA' in config:
+    # column_name specified option
+    if isinstance(config["ILLUMINA"], str):
+        import pandas as pd
+        from glob import glob
+        col_name = config["ILLUMINA"]
+        md_df = pd.read_table(metadata)
+        if not col_name in md_df.columns:
+            raise TypeError('ERROR in ', config_path, ': column name specified for ILLUMINA does not exist in metadata sheet. Please, complete ', config_path)
+        for sampleid in md_df[col_name].to_list():
+            sample_pattern = "{}/{}*".format(raw_dir, sampleid)
```

Comment:
If `S1` and `S11` are listed but only `S11` is present, then it will succeed for both. Could we handle this?